internal/pkg/config/cache.go

```diff
@@ -28,9 +28,7 @@ type Cache struct {
 	APIKeyJitter time.Duration `config:"jitter_api_key"`
 }
 
-func (c *Cache) InitDefaults() {
-	c.LoadLimits(loadLimits(0))
-}
+func (c *Cache) InitDefaults() {}
```

Comment:
tldr: loading the defaults here causes bugs

The `Cache` and `ServerLimits`'s `InitDefaults` methods have been purposefully changed to `nop`s as they may have been ignoring a bug that the TestConfig/input-config test was not catching because the agent limits in default/gt10000_limits.yml was incorrectly specified.

The test was incorrectly generating the default config output because the agent count was not included in the bounds of the limits.yml files. (note that the test to use agent count derived limits instead of memory limits was moved to the "config specifies agent count")

[`Config.Unpack`](https://pkg.go.dev/github.com/elastic/go-ucfg#Config.Unpack) calls `InitDefaults` on the struct (our top level config) and all children that have an `InitDefaults` function (this leads to extra initialization calls which is [another issue](https://github.com/elastic/fleet-server/issues/3090)).

If the `Cache` or `ServerLimits` loads defaults with the `InitDefaults` function then the `LoadLimits` calls with an agent count will not set the limits correctly as the config does not have zero values.

Without the change:
1. Input yml is read and `Unpack` is called. the config struct is loaded with data from the yml.
2. `Unpack` calls `InitDefaults`, which calls `c.LoadLimits(loadLimits(0))` resulting cache/limit values being set to a number that reflects the instance memory.
3. `cfg.LoadServerLimits` is called to use the `limits.max_agents` value that's part of the input
4. envLimits that correspond to the agent limit are found
5. `LoadLimits` compares the config (with memory limit values) and envLimits and does not replace anything.

Using a `nop` means that after `Unpack`, `Server.Limits` and `Server.Cache` are just copies of whatever is explicitly specified in the input, so when `LoadServerLimits` is called it can replace the 0 values correctly